### ISSUE TITLE
Alarm status doesn't update after application update

### ISSUE THREAD

**Issue by Para-lyzed** [sweakpl/qralarm-android#12]
If the user had an alarm set, then they update the app to a newer version, the app no longer shows the alarm as set, but the alarm will still go off. When the alarm goes off, the screen to turn it off does not appear and the user must find and open the app, "start" the alarm, then subsequently stop the alarm immediately afterwards.

For reference, I am running QRAlarm on a OnePlus 7 Pro with Android 11, LineageOS build 18.1-20221009-nightly-guacamole. I doubt this is a device or ROM specific issue, but I could see the behavior potentially differing between major android releases.

**Comment by sweakpl** [comment]
I've tried replicating the issue on Android 7 (Huawei device) and Android 13 (Pixel device) with no success. I've tried the following:

* Installed `1.1.1` version and set an alarm,
* Updated the app to the `1.2` version,
* Tried both of the following:
    * Opened the app immediately after updating:
        * The screen shown that the alarm was set and later it actually started.
    * Did not open the app after the update:
        * The alarm has started later and the the screen with the *Stop* button appeared.

Maybe this might actualy be an issue of how the LineageOS ROM handles app updates (potentially rewriting user preferences) but I don't want to blame Your device or anything like that. Either way, I hope that despite the bug, QRAlarm continues to work as expected.

**Comment by Para-lyzed** [comment]
I have done some testing, and have not been able to replicate the issue. I will provide all the details I can about this issue, but it is very inconsequential, so I am not especially concerned about it.

I have been using the same installation of QRAlarm since version 1.0. It was installed very shortly after release. I know that I set my alarm yesterday for 6AM, and between yesterday morning and 6AM PST today, I had updated QRAlarm. I read the patch notes and was happy to see the gentle wake up feature added, so I opened the app immediately after the update. At that point, the alarm either did not appear as set, or I disabled it after the update (I cannot remember which) since I was able to access the settings and set the gentle wake up for 30 seconds. The alarm did in fact use the gentle wake up when it went off, even though it was set before that feature was added. I use the preset "Alarm clock" sound, a custom QR code, 10 minute snooze, but maximum snoozes set to 0. I updated the app from version 1.1.1 to version 1.2 using Neo Store v0.9.7. I believe that the original install was from Droid-ify, but that should be of no consequence.

I believe the issue may have happened for one of the following reasons:

* The app may have been killed by OnePlus's hyper aggressive battery saving mode, which sometimes still kills apps even when they are set to not be optimized. I believe this issue persisted from OxygenOS into the LineageOS builds. While the app was killed, it was updated, and as such may have had an issue detecting an alarm for some reason.
* The update from Neo Store may have caused some issue either in download or installation.
* Perhaps there was some gradual change to how values were stored between version 1 and version 1.2 that could have caused an error.
* Android 11 or LineageOS itself may have had some odd interaction when the package manager installed the new version.

Whatever the case may be, it seems that the issue can't be (easily) replicated. As such, I will close this issue as a weird one-off interaction.
